src/edi_energy_scraper/__init__.py

```diff
@@ -368,3 +387,62 @@ def get_edifact_version_from_filename(path: Path) -> EdifactFormatVersion:
     format_version = get_edifact_format_version(berlin_local_time)
 
     return format_version
+
+
+def get_publication_date_from_filename(path: Path) -> datetime.datetime:
+    """
+    Determines the publication date of a given file.
+    """
+    filename = path.stem
+    date_string = filename.split("_")[-1]  # Assuming date is in the last part of filename
```

Comment:
Hier könnte man die Variable oder zumindest den Kommentar noch genauer beschreiben. Das letzte Datum ist das "gültig ab" richtig? Jetzt ist ja das "gültig bis" auch relevant ;)

Reply:
done in https://github.com/Hochfrequenz/edi_energy_scraper/commit/5cea45d09cc863f419164b9c4540dad08c098bb3